server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -62,7 +62,15 @@ protected Delegating(ActionListener<DelegateResponse> delegate) {
 
         @Override
         public void onFailure(Exception e) {
-            delegate.onFailure(e);
+            try {
+                delegate.onFailure(e);
+            } catch (RuntimeException ex) {
+                if (ex != e) {
+                    ex.addSuppressed(e);
+                }
+                assert false : new AssertionError("map: listener.onFailure failed", ex);
```

Comment:
Same comment as below, perhaps just:

```suggestion
                assert false : new AssertionError("listener.onFailure failed", ex);
```

Reply:
++